akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpRequestParser.scala

```diff
@@ -180,15 +180,17 @@ private[http] final class HttpRequestParser(
             if (rawRequestUriHeader) `Raw-Request-URI`(uriBytes.decodeString(HttpCharsets.`US-ASCII`.nioCharset)) :: headers
             else headers
 
-          val allHeaders =
+          val requestStart =
             if (method == HttpMethods.GET) {
               Handshake.Server.websocketUpgrade(headers, hostHeaderPresent, websocketSettings, headerParser.log) match {
-                case OptionVal.Some(upgrade) => upgrade :: allHeaders0
-                case OptionVal.None          => allHeaders0
+                case OptionVal.Some(upgrade) =>
+                  RequestStart(method, uri, protocol, Map(AttributeKeys.webSocketUpgrade -> upgrade), upgrade :: allHeaders0, createEntity, expect100continue, closeAfterResponseCompletion)
```

Comment:
So, we would have both the header and the attribute? I guess that's what we want for compatibility?

Reply:
I think so, yes - we can drop the header in 10.3 I guess?